app/controllers/packages_controller.rb

```diff
@@ -0,0 +1,78 @@
+class PackagesController < ApplicationController
+  skip_before_filter :verify_authenticity_token, only: [:save]
+
+  def estimate_request # api endpoint
+      @betsy_shipping = params
+      origin
+      destination
+      package
+
+      begin
+        ups = ActiveShipping::UPS.new(:login => ENV["ACTIVESHIPPING_UPS_LOGIN"],
+                                      :password => ENV["ACTIVESHIPPING_UPS_PASSWORD"],
+                                      :key => ENV["ACTIVESHIPPING_UPS_KEY"])
+        ups_response = ups.find_rates(@origin, @destination, @package)
+        ups_rates = ups_response.rates.sort_by(&:price).collect { |rate| [rate.service_name, rate.price, rate.delivery_date] }
+        usps = ActiveShipping::USPS.new(:login => ENV["ACTIVESHIPPING_USPS_LOGIN"])
+        usps_response = usps.find_rates(@origin, @destination, @package)
+        usps_rates = usps_response.rates.sort_by(&:price).collect { |rate| [rate.service_name, rate.price] }
+
+        carrier_rates = []
+        carrier_rates.push(ups_rates, usps_rates)
+        render json: carrier_rates.as_json
+      rescue ActiveShipping::ResponseError => error
+       error_json = { message: error.message }
+       render json: error_json.as_json, status: 400
+      end
+  end
+
+  def save
+    service_array = params["order"]["estimate"]["service"].split(" ")
+    price = service_array.pop.to_f
+    service_type = service_array.join(" ")
+    weight = params["order"]["estimate"]["service_info"]["estimate"]["total_weight"].to_i
+    @box_size = params["order"]["estimate"]["service_info"]["estimate"]["box_size"]
```

Comment:
Yeah, we definitely wanted to refactor this chunk of code, but ran out of time. Thanks for going through this part even though it might have been a little confusing.